saleor/webhook/transport/utils.py

```diff
@@ -409,42 +412,62 @@ def attempt_update(
     attempt.response_status_code = webhook_response.response_status_code
     attempt.request_headers = json.dumps(webhook_response.request_headers)
     attempt.status = webhook_response.status
-    attempt.save(
-        update_fields=[
-            "duration",
-            "response",
-            "response_headers",
-            "response_status_code",
-            "request_headers",
-            "status",
-        ]
-    )
+
+    if attempt.id:
+        attempt.save(
+            update_fields=[
+                "duration",
+                "response",
+                "response_headers",
+                "response_status_code",
+                "request_headers",
+                "status",
+            ]
+        )
 
 
 @allow_writer()
 def clear_successful_delivery(delivery: "EventDelivery"):
-    if delivery.status == EventDeliveryStatus.SUCCESS:
-        payload_id = delivery.payload_id
-        delivery.delete()
-        if payload_id:
-            payloads_to_delete = EventPayload.objects.filter(
-                pk=payload_id, deliveries__isnull=True
+    if not delivery.id or delivery.status != EventDeliveryStatus.SUCCESS:
+        return
+
+    payload_id = delivery.payload_id
+    delivery.delete()
+    if payload_id:
+        payloads_to_delete = EventPayload.objects.filter(
+            pk=payload_id, deliveries__isnull=True
+        )
+        files_to_delete = [
+            event_payload.payload_file.name
+            for event_payload in payloads_to_delete.using(
+                settings.DATABASE_CONNECTION_REPLICA_NAME
             )
-            files_to_delete = [
-                event_payload.payload_file.name
-                for event_payload in payloads_to_delete.using(
-                    settings.DATABASE_CONNECTION_REPLICA_NAME
-                )
-                if event_payload.payload_file
-            ]
-            payloads_to_delete.delete()
-            delete_files_from_private_storage_task(files_to_delete)
+            if event_payload.payload_file
+        ]
+        payloads_to_delete.delete()
+        delete_files_from_private_storage_task(files_to_delete)
 
 
 @allow_writer()
 def delivery_update(delivery: "EventDelivery", status: str):
     delivery.status = status
-    delivery.save(update_fields=["status"])
+    if delivery.id:
+        delivery.save(update_fields=["status"])
+
+
+@allow_writer()
+def save_unsuccessful_delivery_attempt(attempt: "EventDeliveryAttempt"):
```

Comment:
I suggest renaming the function so it's clearer that it's a conditional save:

```suggestion
def maybe_save_unsuccessful_delivery_attempt(attempt: "EventDeliveryAttempt"):
```

Reply:
But condition is based on status and it skips save only if it's sucessful so I guess the name is fine.
We can consider `maybe_save_delivery_attempt` or `save_delivery_attempt_if_not_sucessful`